tracker/cli/setup.py

```diff
@@ -0,0 +1,96 @@
+from click import echo, option
+from config import basedir, TRACKER_PASSWORD_LENGTH_MIN, TRACKER_PASSWORD_LENGTH_MAX
+from getpass import getpass
+from re import match, IGNORECASE
+from pathlib import Path
+from os.path import isfile
+from os import rename
+from . import cli
+
+
+@cli.group()
+def setup():
+    """Setup and bootstrap the application."""
+    pass
+
+
+@setup.command()
+@option('--purge', is_flag=True, help='Purge all data and tables.')
+def bootstrap(purge=False):
+    """Bootstrap the environment."""
+
+    from tracker import db
+
+    def mkdir(path):
+        Path(path).mkdir(parents=True, exist_ok=True)
+
+    mkdir('{}/pacman/cache'.format(basedir))
+    mkdir('{}/pacman/log'.format(basedir))
+    mkdir('{}/pacman/arch/x86_64/db'.format(basedir))
+
+    if isfile('{}/app.db'.format(basedir)):
+        rename('{}/app.db'.format(basedir),
+               '{}/tracker.db'.format(basedir))
+
+    if purge:
+        echo('Purging the database...')
+        db.drop_all()
+        _vacuum()
+
+    echo('initializing the database...')
+    db.create_all()
+
+
+@setup.command()
+def user():
+    """Create a new user."""
+    from tracker import db
+    from tracker.model.user import User, username_regex
+    from tracker.model.enum import UserRole
+    from tracker.user import random_string, hash_password
+
+    user = User()
+    user.active = True
+
+    print('Username: ', end='')
+    user.name = input()
+    if not user.name or not match(username_regex, user.name) or len(user.name) > User.NAME_LENGTH:
+        print('ERROR: Invalid username')
+        exit(1)
+
+    user.password = getpass()
+    if not user.password:
+        user.password = random_string()
+        print('Generated password: {}'.format(user.password))
+    if len(user.password) > TRACKER_PASSWORD_LENGTH_MAX or len(user.password) < TRACKER_PASSWORD_LENGTH_MIN:
+        print('ERROR: Password must be between {} and {} characters.'
+              .format(TRACKER_PASSWORD_LENGTH_MIN, TRACKER_PASSWORD_LENGTH_MAX))
+        exit(1)
+    user.salt = random_string()
+    user.password = hash_password(user.password, user.salt)
+
+    print('E-Mail: ', end='')
+    user.email = input()
+    if not match(r'^.+@([^.@][^@]+)$', user.email, IGNORECASE):
+        print('ERROR: Invalid E-Mail')
+        exit(1)
+
+    print('Role: ', end='')
+    user.role = UserRole.fromstring(input())
```

Comment:
Still wondering how a user knows what roles are available when running this command.

Reply:
added exposure in click so:
```./trackerctl setup user --help```
will help you
otherwise the prompt will also show valid options when failing with non-default

Reply:
Nice!